packages/caliper-core/test/worker/rate-control/recordRate.js

```diff
@@ -63,42 +71,263 @@ describe('RecordRate controller', () => {
             totalWorkers: 2
         };
 
-        const testMessage = new TestMessage('test', [], msgContent);
-        const stubStatsCollector = sinon.createStubInstance(TransactionStatisticsCollector);
-        const rateController = RecordRate.createRateController(testMessage, stubStatsCollector, 0);
-        const mockRate = MockRate.createRateController();
-        mockRate.reset();
-        mockRate.isApplyRateControlCalled().should.equal(false);
-        await rateController.applyRateControl();
-        mockRate.isApplyRateControlCalled().should.equal(true);
+        stubStatsCollector = new TransactionStatisticsCollector();
+        stubStatsCollector.getTotalSubmittedTx = sandbox.stub();
     });
 
-    it('should throw an error if the rate controller to record is unknown', async () => {
-        const msgContent = {
-            label: 'test',
-            rateControl: {
-                "type": "record-rate",
-                "opts": {
-                    "rateController": {
-                        "type": "nonexistent-rate"
-                    },
-                    "pathTemplate": "../tx_records_client<C>_round<R>.txt",
-                    "outputFormat": "TEXT",
-                    "logEnd": true
-                }
-            },
-            workload: {
-                module: 'module.js'
-            },
-            testRound: 0,
-            txDuration: 250,
-            totalWorkers: 2
-        };
-        const testMessage = new TestMessage('test', [], msgContent);
+    afterEach(() => {
+        sandbox.restore();
+    });
+
+    describe('happy path', () => {
```

Comment:
Sorry I don't like this description in the describe. The it statement provide details on what the final outcome should be. This describe block should collate the tests under a common group of activities. I'm more than happy for multiple describes to split this up and also that these describes can contain error scenarios as well. For example, the testing of the export formats can be done under their own describe. It may also make sense not to have some of the tests under a describe. See the fabric tests for examples.